test/kitchen/test/integration/system-probe-test/rspec_datadog/system-probe-test_spec.rb

```diff
@@ -13,16 +13,41 @@
   "pkg/collector/corechecks/ebpf/probe"
 )
 
+# list based on `find pkg -name "*_test.go" | xargs grep -l "linux_bpf" | xargs dirname | sort -u`
 runtime_compiled_tests = Array.[](
-  "pkg/network/tracer",
+  "pkg/collector/corechecks/ebpf/probe",
+  "pkg/ebpf",
+  "pkg/ebpf/bytecode/runtime",
+  "pkg/ebpf/compiler",
+  "pkg/network",
+  "pkg/network/dns",
+  "pkg/network/netlink",
+  "pkg/network/protocols",
+  "pkg/network/protocols/events",
+  "pkg/network/protocols/grpc",
   "pkg/network/protocols/http",
-  "pkg/collector/corechecks/ebpf/probe"
+  "pkg/network/tracer",
+  "pkg/network/tracer/connection",
+  "pkg/network/tracer/connection/kprobe",
+  "pkg/network/usm",
 )
 
 co_re_tests = Array.[](
-  "pkg/network/tracer",
+  "pkg/collector/corechecks/ebpf/probe",
+  "pkg/ebpf",
+  "pkg/ebpf/bytecode/runtime",
+  "pkg/ebpf/compiler",
+  "pkg/network",
+  "pkg/network/dns",
+  "pkg/network/netlink",
+  "pkg/network/protocols",
+  "pkg/network/protocols/events",
+  "pkg/network/protocols/grpc",
   "pkg/network/protocols/http",
-  "pkg/collector/corechecks/ebpf/probe"
+  "pkg/network/tracer",
+  "pkg/network/tracer/connection",
+  "pkg/network/tracer/connection/kprobe",
+  "pkg/network/usm",
```

Comment:
So a lot of these packages don't need to run in multiple modes. I've got a branch nearly ready that solves this problem. Can you limit this PR to your original intended scope of just `pkg/network/usm`?

Reply:
yes and no :)
what's the rule here ?  linux_bpf tests are running only on sysprobe CI, so probably should be tested multiple kernels/distro/...

* What your trick to filter these ?
tests packages that start a tracer of an ebpf program should be listed here, but this require to filter by hand
* Even if they will be executed multiple time and not starting ebpf, the time consumed by these tests is minimal
ex: TestGetAvailableHelpers shall be tested on different distro (apt, yum, kernels)
* How we sure we don't miss something in the future ?

Reply:
> What your trick to filter these

I'm moving that decision into the individual tests themselves. We already have a really long test runtime, so we should only run as many tests as we need and nothing more.

> TestGetAvailableHelpers shall be tested on different distro

It already was. It just doesn't need to be run for prebuilt, runtime compilation, and CO-RE because the test doesn't depend on the build mode.

Reply:
Adding every package that uses the `linux_bpf` build tag is out of scope for this PR. Can you focus it back down to the original purpose?

Reply:
I got confused with prebuilt/runtime/core and distro mapped
I did a pass only for network and re selected the packages

Solution on per test basis looks good